packages/@lwc/engine-core/src/framework/template.ts

```diff
@@ -343,9 +314,14 @@ function buildParseFragmentFn(
 
             htmlFragment += strings[strings.length - 1];
 
-            cache[cacheKey] = createFragmentFn(htmlFragment, renderer);
+            const element = createFragmentFn(htmlFragment, renderer);
+
+            // Cache is only here to prevent calling innerHTML multiple times which doesn't happen on the server.
+            if (process.env.IS_BROWSER) {
+                setInFragmentCache(cacheKey, strings, element);
+            }
```

Comment:
Small bug I noticed – the fragment cache is only ever used in the browser, but we were still (uselessly) setting it here even on the server.

Reply:
🤔 Would it make sense to put the browser check within the function so that we don't forget about it in other places?

Reply:
Done: 3fdcaf4ff